internal/types/utils.go

```diff
@@ -67,6 +67,17 @@ func StringSliceContains(slice []string, value string) bool {
 	return false
 }
 
+// Checks value occurs the expected number of times in the slice
+func ValidateStringCountInSlice(slice []string, value string, expectedCount int) (bool, int) {
```

Comment:
I think this function would make a little more sense as just a counting function, that just returns the number of times that string was found. It's easier to understand reading the code if this just returns a number and wherever it is called does the check for if that equals zero or whatever.

If you switch to using a map then you may not need this method at all.